source/main/dqm4hep-mysql-push-parameters.cc

```diff
@@ -0,0 +1,201 @@
+    return 1;
+  }
+  
+  for(TiXmlElement *db = databases->FirstChildElement("db") ; db ; db = db->NextSiblingElement("db"))
+  {
+    const char *dbID = db->Attribute("id");
+    
+    if(!dbID || std::string(dbID).empty())
+    {
+      dqm_error( "ERROR: <db> tag has no 'id' attribute or is empty, line {0}", db->Row() );
+      return 1;
+    }
+    
+    if(dbInterfaceMap.find(dbID) != dbInterfaceMap.end())
+    {
+      dqm_error( "ERROR: <db> tag with id '{0}', line {1}, is duplicated !", dbID, db->Row() );
+      return 1;
+    }
+    
+    std::string dbHost = db->Attribute("host") ? db->Attribute("host") : "localhost";
+    std::string dbUser = db->Attribute("user") ? db->Attribute("user") : "DQM4HEP";
+    std::string dbDb = db->Attribute("db") ? db->Attribute("db") : "DQM4HEP";    
+    std::string dbPassword = db->Attribute("password") ? db->Attribute("password") : "";
+
+    DBInterfaceMetadata metadata;
+    metadata.m_dbInterface = std::make_shared<DBInterface>();
+    metadata.m_host = dbHost;
+    metadata.m_user = dbUser;
+    metadata.m_db = dbDb;
+    metadata.m_password = dbPassword;
+    
+    dbInterfaceMap[dbID] = metadata; 
+  }
+  
+  int nParameters(0);
+  std::set<std::pair<std::string,std::string>> dbTables;
+  
+  for(TiXmlElement *parameters = root->FirstChildElement("parameters") ; parameters ; parameters = parameters->NextSiblingElement("parameters"))
+  {
+    std::string dbID = parameters->Attribute("db") ? parameters->Attribute("db") : "";
+    std::string dbTable = parameters->Attribute("table") ? parameters->Attribute("table") : "";
+    
+    auto findIter = dbInterfaceMap.find(dbID);
+    
+    if(dbInterfaceMap.end() == findIter)
+    {
+      dqm_error( "ERROR: <parameters> tag, db attribute '{0}' not found in <databases> section, line {1}", dbID, parameters->Row() );
+      return 1;
+    }
+    
+    if(!findIter->second.m_dbInterface->isConnected())
+    {
+      if(findIter->second.m_password.empty())
+      {
+        std::stringstream passwordText;
+        passwordText << "Password [host=" << findIter->second.m_host << " db=" << findIter->second.m_db << " user=" << findIter->second.m_user << "] : ";
+        char *password = getpass(passwordText.str().c_str());
+        
+        if(!password)
+        {
+          dqm_error( "ERROR: Invalid password !" );
+          return 1;
+        }
+        
+        findIter->second.m_password = password;
+      }
+      
+      // connect to db
+      THROW_RESULT_IF(STATUS_CODE_SUCCESS, !=, 
+        findIter->second.m_dbInterface->connect(
+          findIter->second.m_host, 
+          findIter->second.m_user, 
+          findIter->second.m_password, 
+          findIter->second.m_db)
+        );
+    }
+    
+    StringMap parameterMap;
+    
+    for(TiXmlElement *parameter = parameters->FirstChildElement("parameter") ; parameter ; parameter = parameter->NextSiblingElement("parameter"))
```

Comment:
This time user will get an error from the throw at [Line186](https://github.com/DQM4HEP/dqm4hep-core/pull/17/files#diff-4aaae144f0df7fd010d08273ba86cefdR186)